catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/DefaultAttributeValueRegistry.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.data;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+/**
+ * Manages default attribute values that apply to all metacard types and default attribute values
+ * that apply to specific metacard types.
+ */
+public interface DefaultAttributeValueRegistry {
+    /**
+     * Registers the given value as the default value for the given attribute across all metacard
+     * types. Overwrites the current default value for the attribute if one exists.
+     *
+     * @param attributeName the name of the attribute, cannot be null
+     * @param defaultValue  the default value, cannot be null
+     * @throws IllegalArgumentException if either argument is null
+     */
+    void setDefaultValue(String attributeName, Serializable defaultValue);
```

Comment:
From a contractual point of view, what happens if I call:
1. `setDefaultValue("foobar", "baz");`
2. `setDefaultValue("myspecialmetatype", "foobar", "special snowflake");

How does that affect the semantics of the get and remove methods?